File: scratch/restore_pvp_energy.py
import urllib.request
import json
import time

url = "https://yetzjqqnmllwufmzopor.supabase.co/rest/v1/profiles"
key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InlldHpqcXFubWxsd3VmbXpvcG9yIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODI3NTkwMzgsImV4cCI6MjA5ODMzNTAzOH0.Ra2mdK9QS4Aq5WZsUmULvqfdaJkdLJBcEzPch9EpwB4"

headers = {
    "apikey": key,
    "Authorization": f"Bearer {key}",
    "Content-Type": "application/json"
}

print("Fetching profiles...")
req = urllib.request.Request(url, headers=headers)
try:
    with urllib.request.urlopen(req) as response:
        rows = json.loads(response.read().decode('utf-8'))
        target = None
        for r in rows:
            if r.get('data') and r['data'].get('username') and r['data']['username'].lower() == 'adminus':
                target = r
                break
        
        if target:
            wallet = target['wallet_address']
            profile_data = target['data']
            profile_data['pvpEnergy'] = profile_data.get('pvpEnergyMax', 5)
            profile_data['lastPvpEnergyRefill'] = int(time.time() * 1000)
            
            patch_url = f"{url}?wallet_address=eq.{wallet}"
            patch_data = json.dumps({"data": profile_data, "updated_at": "now()"}).encode('utf-8')
            
            patch_req = urllib.request.Request(patch_url, data=patch_data, headers=headers, method='PATCH')
            with urllib.request.urlopen(patch_req) as patch_response:
                print(f"Successfully restored PvP energy to {profile_data['pvpEnergy']} for Adminus!")
        else:
            print("Adminus not found. Registered usernames:")
            for r in rows:
                if r.get('data') and r['data'].get('username'):
                    print("-", r['data']['username'])
except Exception as e:
    print("Error:", e)

File: scratch/restore_pvp_energy.js
const { createClient } = require('@supabase/supabase-js');
const supabaseUrl = process.env.VITE_SUPABASE_URL || 'https://yetzjqqnmllwufmzopor.supabase.co';
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.VITE_SUPABASE_ANON_KEY || 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InlldHpqcXFubWxsd3VmbXpvcG9yIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODI3NTkwMzgsImV4cCI6MjA5ODMzNTAzOH0.Ra2mdK9QS4Aq5WZsUmULvqfdaJkdLJBcEzPch9EpwB4';
const supabase = createClient(supabaseUrl, supabaseKey);

async function run() {
  console.log('Searching for user "adminus"...');
  const { data: rows, error } = await supabase
    .from('profiles')
    .select('id, wallet_address, data');

  if (error) {
    console.error('Failed to query profiles:', error);
    return;
  }

  const target = rows.find(r => r.data && r.data.username && r.data.username.toLowerCase() === 'adminus');
  if (target) {
    console.log(`Found user: ${target.data.username} (Wallet: ${target.wallet_address})`);
    const updatedData = {
      ...target.data,
      pvpEnergy: target.data.pvpEnergyMax || 5,
      lastPvpEnergyRefill: Date.now()
    };

    const { data: updateRes, error: updateErr } = await supabase
      .from('profiles')
      .update({ data: updatedData, updated_at: new Date().toISOString() })
      .eq('wallet_address', target.wallet_address)
      .select('wallet_address');

    if (updateErr) {
      console.error('Failed to update energy:', updateErr);
    } else {
      console.log(`Successfully restored PvP energy to ${updatedData.pvpEnergy} for ${target.data.username}!`);
    }
  } else {
    console.log('User "adminus" not found. Registered usernames list:');
    rows.forEach(r => {
      if (r.data && r.data.username) {
        console.log(`- ${r.data.username} (${r.wallet_address})`);
      }
    });
  }
}
run();
